### ISSUE TITLE
Continuous transmission causes audio not to be sent on startup

### ISSUE THREAD

**Issue by wibbly** [mumble-voip/mumble#1140]
When I connect my OSX client (1.2.4-223-g29a65c6) to my murmur server, although it appears to indicate it is sending audio (red lips showing), other clients connected to the server don't receive any audio. They see black lips.

If I mute and unmute the client it starts sending.

This happens when my audio sending mode is set to continuous at startup. If I use voice activated mode and a high threshold it's fine (perhaps because it's not 'on' at startup). If I set that threshold very low (so it behaves like 'continuous') the audio doesn't ever start sending.

So ... it looks like there is some problem beginning the audio sending if the application starts up with sending enabled.

Muting and unmuting, or using push-to-talk, or setting a high threshold and using voice activation all work around the problem. It seems that it's only problematic if the audio begins 'on' and doesn't get turned on and off.

**Comment by Krzmbrzl** [comment]
We're you able to reproduce this again with Mumble 1.3?

**Comment by ancarda** [comment]
I've attempted to reproduce this (Mumble 1.3.3 on macOS 10.15.7). No matter what I try, everyone can hear me if I launch with continuous transmission. This applies both if I went through the server selection UI at startup or if I launched directly into a channel using [a mumble:// URI](https://wiki.mumble.info/wiki/Mumble_URL).

@wibbly: Is this still an issue for you? I can try to look into a fix for you if it is, but I will need a way to reproduce the problem on my end

**Comment by Krzmbrzl** [comment]
_Maybe_ this could be related to #4547 ?

**Comment by ancarda** [comment]
Because I use Mumble a lot, it already had microphone access when I tried to reproduce this bug. It could be something like:

1) Mumble tries to get an audio stream but gets a "null stream" because it doesn't have that permission
2) After being granted access, Mumble isn't aware that it's been granted (not listening to the event?) and should try again, thus the user has to manually poke Mumble to try again (i.e. some action like mute and unmute).

However, that feels like a stretch as this issue is worded like it's a recurring problem; not an issue you'd run into the first time you use Mumble.

Nevertheless, I'll revoke the microphone permission and try to reproduce this again.

**Comment by ancarda** [comment]
I closed Mumble and revoked microphone permission (Security & Privacy pane in System Preferences). Nobody could hear me when I started up Mumble, despite my icon lighting up. However, when I switched to push-to-talk, people still could not hear me even though my icon was lighting up. So, I still don't know how to reproduce this bug, as the initial bug report suggested that would work.

Having said that, I was never promoted by macOS to give Microphone access, as I think unchecking in System Preferences changes it to "always-deny" rather than "ask-for-access". It may be worth detecting this state (if possible?) and showing a warning that the user needs to manually give Mumble access for the microphone to work correctly. Is it okay if I make a separate issue for that?

I can reinstall Mumble if anyone wants me to check what happens with the "ask-for-access" state, as it may well be sub-optimal, but I suspect the privacy permissions aren't to blame for this particular issue.
